pkg/cmd/cli/cmd/loginoptions.go

```diff
@@ -439,3 +439,23 @@ func (o *LoginOptions) serverProvided() bool {
 func (o *LoginOptions) tokenProvided() bool {
 	return len(o.Token) > 0
 }
+
+func (o *LoginOptions) ConfigIsWritable() (bool, error) {
+	kubeconfig := o.PathOptions.GetDefaultFilename()
+
+	info, err := os.Stat(kubeconfig)
+	if err != nil {
+		if os.IsNotExist(err) {
+			info, err = os.Stat(filepath.Dir(kubeconfig))
+			if err != nil {
+				return false, err
+			}
+		} else if os.IsPermission(err) {
+			return false, nil
+		} else {
+			return false, err
+		}
+	}
+
+	return info.Mode().Perm()&0200 != 0, nil
```

Comment:
> It seems like we shouldn't be trying to replicate the authz rules for file access -- why don't we just try to open it for writing and see if that works? Will doing the uid/gid check work for all cases? What if the filesystem is mounted read-only?

It'll work for more cases than this.  I'm not sure if a request for opening to append and appending nothing works.  If it does, then that would do.